sdk/servicebus/azure_messaging_servicebus/src/client.rs

```diff
@@ -0,0 +1,559 @@
+};
+use azure_core::{credentials::TokenCredential, fmt::SafeDebug, http::ClientOptions, http::Url};
+use azure_core_amqp::{
+    AmqpConnection, AmqpConnectionApis, AmqpConnectionOptions, AmqpOrderedMap, AmqpSymbol,
+    AmqpValue,
+};
+use std::sync::Arc;
+
+/// SubQueue allows you to target a subqueue of a queue or subscription.
+///
+/// For example, the dead letter queue (SubQueueDeadLetter) or transfer dead letter queue (SubQueueTransfer).
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum SubQueue {
+    /// Targets the dead letter queue for a queue or subscription.
+    DeadLetter,
+    /// Targets the transfer dead letter queue for a queue or subscription.
+    Transfer,
+}
+
+impl SubQueue {
+    /// Returns the path suffix for the sub-queue.
+    pub fn as_path_suffix(&self) -> &'static str {
+        match self {
+            SubQueue::DeadLetter => "/$DeadLetterQueue",
+            SubQueue::Transfer => "/$Transfer/$DeadLetterQueue",
+        }
+    }
+}
+
+/// Options for configuring a Service Bus client.
+#[derive(Clone, SafeDebug)]
+pub struct ServiceBusClientOptions {
+    /// The API version to use when communicating with the Service Bus service.
+    pub api_version: String,
+
+    /// Core client configuration options.
+    pub client_options: ClientOptions,
+
+    /// Application ID that will be passed to the namespace.
+    ///
+    /// This optional identifier is passed to the Service Bus namespace during connection establishment
+    /// and can be used for diagnostic purposes. It follows the same pattern as the Go SDK's ApplicationID.
+    pub application_id: Option<String>,
+}
+
+impl Default for ServiceBusClientOptions {
+    fn default() -> Self {
+        Self {
+            api_version: "2017-04".to_string(), // Default Service Bus API version
+            client_options: ClientOptions::default(),
+            application_id: None,
+        }
+    }
+}
+
+/// Options for creating a sender.
+#[derive(Clone, Default)]
+pub struct CreateSenderOptions; // Place holder for future options
+
+/// Options for creating a receiver.
+#[derive(Clone)]
+pub struct CreateReceiverOptions {
+    /// The receive mode for the receiver.
+    pub receive_mode: ReceiveMode,
+    /// The sub-queue to target (e.g., dead letter queue).
+    pub sub_queue: Option<SubQueue>,
+}
+
+impl Default for CreateReceiverOptions {
+    fn default() -> Self {
+        Self {
+            receive_mode: ReceiveMode::PeekLock,
+            sub_queue: None,
+        }
+    }
+}
+
+/// A client for interacting with Azure Service Bus.
+pub struct ServiceBusClient {
+    connection: Arc<AmqpConnection>,
+    namespace: String,
+    options: ServiceBusClientOptions,
+    authorizer: Option<Arc<Authorizer>>,
+}
+
+impl ServiceBusClient {
+    /// Creates a helper function to build AmqpConnectionOptions from ServiceBusClientOptions.
+    fn build_connection_options(
+        options: Option<ServiceBusClientOptions>,
+    ) -> Option<AmqpConnectionOptions> {
```

Comment:
This is not a good pattern. Either include them on the options bag and/or in the builder. That's what they are there for. If you want the builder to use either a credential OR connection string, use a typestate builder as documented: https://azure.github.io/azure-sdk/rust_introduction.html#rust-builders